external-crates/move/crates/move-analyzer/src/symbols/compilation.rs

```diff
@@ -239,54 +241,59 @@ pub fn get_compiled_pkg(
                 .collect::<BTreeSet<_>>();
 
             let pkg_info = packages_info.lock().unwrap();
-            let (pkg_cached_deps, edition, compiler_info) = match pkg_info.get(pkg_path) {
-                Some(d)
-                    if manifest_hash.is_some()
-                        && manifest_hash == d.manifest_hash
-                        && deps_hash == d.deps_hash =>
-                {
-                    eprintln!("found cached deps for {:?}", pkg_path);
-                    (
-                        Some(AnalyzedPkgInfo {
-                            program_deps: d.deps.clone(),
-                            symbols_data: Some(d.deps_symbols_data.clone()),
-                            program: Some(d.program.clone()),
-                            file_hashes: d.file_hashes.clone(),
+            let (pkg_cached_deps, edition, compiler_info, compiled_libs) =
+                match pkg_info.get(pkg_path) {
+                    Some(d)
+                        if manifest_hash.is_some()
+                            && manifest_hash == d.manifest_hash
+                            && deps_hash == d.deps_hash =>
```

Comment:
What is this actually checking?

Reply:
This is old code - it checks if dependencies did not change underneath, in which case we have to drop the cache and recompute everything